File: output/xCodeEval_8000/C++/C/f99e4692c10adf6a0c2b573ba530305d.c
#include <stdio.h>

#define N 1000009

int a[N], n, s[N], top;

int main(){

    scanf("%d", &n);
    for(int i=0; i<n; i++){
        scanf("%d", &a[i]);
    }
    int x;
    scanf("%d", &x);
    a[x-1] *= -1;

    for(int i=n-1; i>=0; i--){
        if(a[i] > 0){
            if(top && s[top] == a[i]){
                top--;
            }
            else{
                s[++top] = a[i];
                a[i] = -a[i];
            }
        }
        else{
            s[++top] = -a[i];
        }
    }

    if(top){
        printf("NO\n");
    }
    else{
        printf("YES\n");
        for(int i=0; i<n; i++){
            printf("%d ", a[i]);
        }
        printf("\n");
    }

    return 0;
}

File: output/xCodeEval_8000/C++/Python/f99e4692c10adf6a0c2b573ba530305d.py
N = int(1e6) + 9
a = [0] * N
n = int(input())
a[:n] = list(map(int, input().split()))
s = [0] * N
top = 0

def RD(x):
    return int(x)

def OT(x):
    print(x, end=' ')

for i in range(n):
    a[i] = RD(a[i])
a[RD() - 1] *= -1

for i in range(n-1, -1, -1):
    if a[i] > 0:
        if top and s[top] == a[i]:
            top -= 1
        else:
            top += 1
            s[top] = a[i]
            a[i] = -a[i]
    else:
        top += 1
        s[top] = -a[i]

if top:
    print("NO")
else:
    print("YES")
    for i in range(n):
        OT(a[i])
